weather_sp/splitter_pipeline/file_name_utils.py

```diff
@@ -22,20 +24,56 @@
 NETCDF_FILE_ENDINGS = ('.nc', '.cd')
 
 
-class OutFileInfo(t.NamedTuple):
+@dataclass
+class OutFileInfo:
+    """Holds data required to construct an output file name.
+
+    Attributes:
+        file_name_template: base output path, may contain python-style formatting
+                            marks. This can be a base directory or a full name.
+        formatting: added after file_name_template to add formatting. Only used
+                    when using --output-dir.
+        ending: file ending.
+        template_folders:
+    """
     file_name_template: str
-    ending: str
-    output_dir: bool = False
+    formatting: str = ''
+    ending: str = ''
+    template_folders: t.List[str] = field(default_factory=lambda: [])
```

Comment:
A typical pattern in python is to set mutable data structures to None, and then set it to an empty collection. 

https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

So, in this case, this would be a `t.Optional[t.List[str]]`, and in the `__post_init__` method, you'd check for None and set it to an empty list.

Reply:
Made all parameters required instead, really makes more sense to set them explicitly.